files/default/sql/ddl/3.2.0__initial_tables.sql

```diff
@@ -2311,3 +2311,97 @@ CREATE TABLE IF NOT EXISTS `hdfs_command_execution` (
+    PRIMARY KEY (`id`),
+    KEY (`feature_name`),
+    KEY (`name`),
+    UNIQUE KEY `config_name_UNIQUE` (`name`, `feature_group_id`, `feature_view_id`),
+    CONSTRAINT `fg_monitoring_config_fk` FOREIGN KEY (`feature_group_id`) REFERENCES `feature_group` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+    CONSTRAINT `fv_monitoring_config_fk` FOREIGN KEY (`feature_view_id`) REFERENCES `feature_view` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+    CONSTRAINT `job_monitoring_config_fk` FOREIGN KEY (`job_id`) REFERENCES `jobs` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE = ndbcluster DEFAULT CHARSET = latin1 COLLATE = latin1_general_cs;
+
+CREATE TABLE IF NOT EXISTS `monitoring_window_config` (
+    `id` INT(11) NOT NULL AUTO_INCREMENT,
+    `window_config_type` INT(11) NOT NULL,
+    `specific_id` INT(11),
+    `time_offset` VARCHAR(63),
+    `window_length` VARCHAR(63),
+    `row_percentage` INT(11),
+    `specific_value` FLOAT,
+    PRIMARY KEY (`id`)
+) ENGINE = ndbcluster DEFAULT CHARSET = latin1 COLLATE = latin1_general_cs;
+
+CREATE TABLE IF NOT EXISTS `statistics_comparison_config` (
+    `id` INT(11) NOT NULL AUTO_INCREMENT,
+    `strict` BOOLEAN DEFAULT FALSE,
+    `relative` BOOLEAN DEFAULT FALSE,
+    `threshold` FLOAT,
+    `compare_on` INT(11) NOT NULL,
+    PRIMARY KEY (`id`)
+) ENGINE = ndbcluster DEFAULT CHARSET = latin1 COLLATE = latin1_general_cs;
+
+CREATE TABLE IF NOT EXISTS `feature_monitoring_result` (
+    `id` INT(11) NOT NULL AUTO_INCREMENT,
+    `feature_monitoring_config_id` INT(11) NOT NULL,
+    `execution_id` INT(11) NOT NULL,  -- dummy this should become ref to another table
+    `monitoring_time` timestamp DEFAULT CURRENT_TIMESTAMP,
+    `shift_detected` BOOLEAN DEFAULT FALSE,
+    `detection_stats_id` INT(11) NOT NULL, -- dummy this should become ref to another table
+    `reference_stats_id` INT(11), -- dummy this should become ref to another table
+    `difference` FLOAT,
+    PRIMARY KEY (`id`),
+    CONSTRAINT `config_monitoring_result_fk` FOREIGN KEY (`feature_monitoring_config_id`) REFERENCES `feature_monitoring_config` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE = ndbcluster DEFAULT CHARSET = latin1 COLLATE = latin1_general_cs;
+
+CREATE TABLE IF NOT EXISTS `feature_descriptive_statistics` (
+    `id` int(11) NOT NULL AUTO_INCREMENT,
+    `feature_type` varchar(20) NOT NULL,
+    `min` FLOAT NULL, 
+    `max` FLOAT NULL,   
+    `sum` FLOAT NULL,
+    `count` BIGINT NULL,
+    `mean` FLOAT NULL,  
+    `stddev` FLOAT NULL,
+    `completeness` FLOAT NULL,
+    `num_non_null_values` BIGINT NULL,
+    `num_null_values` BIGINT NULL,
+    `percentiles` BLOB,
+    `approx_num_distinct_values` BIGINT NULL,
+    -- with exactUniqueness
+    `distinctness` FLOAT NULL,
+    `entropy` FLOAT NULL,
+    `uniqueness` FLOAT NULL,
+    `exact_num_distinct_values` BIGINT NULL,
+    -- for filtering
+    `start_time` TIMESTAMP NULL,
+    `end_time` TIMESTAMP NULL,
```

Comment:
Is it possible to have an index over the filtering columns, so we don't have to do a whole table scan?

Reply:
The filtering column is feature_name. Start and end times were moved to a different table.